applications/CoSimulationApplication/python_scripts/coupling_interface_data.py

```diff
@@ -40,12 +36,22 @@ def Initialize(self):
         if not self.variable_type in admissible_scalar_variable_types and not self.variable_type in admissible_vector_variable_types:
             raise Exception('The input for "variable" "{}" is of variable type "{}" which is not allowed, only the following variable types are allowed:\n{}, {}'.format(variable_name, self.variable_type, ", ".join(admissible_scalar_variable_types), ", ".join(admissible_vector_variable_types)))
 
-        self.variable = KM.KratosGlobals.GetVariable(variable_name)
+        self.variable = KM.KratosGlobals.GetVariable(variable_name) # TODO here maybe we could construct a new var if necessary (maybe clashes with delayed app-import ...?)
 
         self.dtype = GetNumpyDataType(self.variable_type) # required for numpy array creation
 
         self.is_scalar_variable = self.variable_type in admissible_scalar_variable_types
 
+        # location of data on ModelPart
+        self.location = self.settings["location"].GetString()
+        admissible_locations = ["node_historical", "node_non_historical", "element", "condition", "process_info", "model_part"]
```

Comment:
Process_info ?? 
What kind of info do you have in mind to put it in Process_info ?? I would avoid using Process_info

Reply:
Also historical and non_historical ... I feel like its a detail that is not relevant for Co-Simulation ! We can restrict it to the historical data what do you say ?

Reply:
I would like to not limit the use-cases beforehand
regarding ProcessInfo: I imagine things like scalar-values coming from e.g. an sdof solver could be saved there
In Kratos many things are saved there, I just don't want to exclude those cases

regarding non-historical: E.g. loads are under discussion to be moved to the [non-hist database](https://github.com/KratosMultiphysics/Kratos/pull/3447) hence this should be possible to be used
Also other discretizations like MPM sometimes save non-historical

=> note that the default is node_historical

Reply:
process-info is removed now, as discussed